packages/kit/src/module/compatibility.ts

```diff
@@ -11,8 +11,20 @@ import { loadNuxtModuleInstance } from './install'
  * that it cannot detect if a module is _going to be_ installed programmatically by another module.
  */
 export function hasNuxtModule (moduleName: string, nuxt: Nuxt = useNuxt()) : boolean {
+  // check installed modules
   return nuxt.options._installedModules.some(({ meta }) => meta.name === moduleName) ||
-    nuxt.options.modules.includes(moduleName)
+    // check modules to be installed
+    nuxt.options.modules
+      .some((m) => {
+        // input may either a string, an array or a module instance
+        function resolveModuleEntry (input: typeof m): boolean {
```

Comment:
Do you think we can hoist this out of the inline function?

Reply:
Good idea :+1: